Propagate environment #1358 (#1392)
seek-oss/skuba: docs/deep-dives/github.md

```

With Docker Compose,
declare volume mounts in your [Compose file].

> You won't need to declare the environment variables in the compose file anymore since `propagate-environment: true` will ensure the variables defined are propagated to the docker environment.

Reviewer: Can we document how a step with the `docker-compose` plugin should look like, similar to the example above for the `docker` plugin?

> You won't need to declare the environment variables in the compose file anymore since `propagate-environment: true` will ensure the variables defined are propagated to the docker environment.

Reviewer 2: Could we include a similar buildkite pipeline example in the same vein as the docker plugin section above?

```

With Docker Compose,
declare volume mounts in your [Compose file].

You won't need to declare the environment variables in the compose file since `propagate-environment: true` will ensure the variables defined are propagated to the docker environment.

For example, with the [Docker Compose Buildkite plugin], we will need to add the `propagate-environment: true`:

```yaml
steps:
  - commands:
      - pnpm run lint
      - pnpm run test
    env:
      # At SEEK, this instructs the build agent to populate the GITHUB_API_TOKEN environment variable for this step.
      GET_GITHUB_TOKEN: 'please'
    plugins:
      - *aws-sm
      - *private-npm
      - *docker-ecr-cache
      - docker#v5.0.0:
          # Enable GitHub integrations.
          environment:
            - GITHUB_API_TOKEN
          propagate-environment: true
          volumes:
            # Mount cached dependencies.
            - /workdir/node_modules
```

And then your [Compose file] will not need to define the `environment` like you had to before to ensure the environment variables are available in Docker

> Docker compose file before:

Reviewer 2: Hmmm. I don't think we have to document the before and after within the deep-dive itself. We can probably get away with just showing what the optimal configuration is.

We can probably move some of the `before` and `after` scenarios to the changeset

```

With Docker Compose,
declare volume mounts in your [Compose file].

You won't need to declare the environment variables in the compose file since `propagate-environment: true` will ensure the variables defined are propagated to the docker environment.

> For example, with the [Docker Buildkite plugin], we will need to add the `propagate-environment: true`:

Reviewer: ```suggestion
For example, with the [Docker Compose Buildkite plugin]:
```

Author: what should this link to? there is no `[Docker Compose Buildkite plugin]` doco linked down here: https://github.com/seek-oss/skuba/blob/master/docs/deep-dives/github.md?plain=1#L123

Reviewer: I think here: https://github.com/buildkite-plugins/docker-compose-buildkite-plugin

Author: updated

```yaml
steps:
  - commands:
      - pnpm run lint
      - pnpm run test
    env:
      # At SEEK, this instructs the build agent to populate the GITHUB_API_TOKEN environment variable for this step.
      GET_GITHUB_TOKEN: 'please'
    plugins:

>       - docker#v5.0.0:
>           # Enable GitHub integrations.
>           environment:
>             - GITHUB_API_TOKEN
>           propagate-environment: true
>           volumes:
>             # Mount cached dependencies.
>             - /workdir/node_modules

Reviewer: ```suggestion
      - *aws-sm
      - *private-npm
      - *docker-ecr-cache
      - docker-compose#v4.16.0:
          # Enable GitHub integrations.
          environment:
            - GITHUB_API_TOKEN
          run: app
          propagate-environment: true
```
